lib/main.js

```diff
@@ -1,33 +1,70 @@
 'use babel';
 
-// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
 import { CompositeDisposable } from 'atom';
-import Path from 'path';
-import * as helpers from 'atom-linter';
 
-// Local vars
-const regex = /(.+):(\d+)\t*(.+)/g;
+let helpers;
+let path;
 
-// Settings
-let executablePath;
-let rulesets;
+function loadDeps() {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+  if (!path) {
+    path = require('path');
+  }
+}
 
 export default {
   activate() {
-    require('atom-package-deps').install('linter-phpmd');
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterPhpmdDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-phpmd');
+      }
+      loadDeps();
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterPhpmdDeps);
+    this.idleCallbacks.add(depsCallbackID);
 
     this.subscriptions = new CompositeDisposable();
+
+    // Backwards compatibility
+    if (atom.config.get('linter-phpmd.rulesets') !== undefined) {
+      this.rulesOrConfigFile = atom.config.get('linter-phpmd.rulesets');
```

Comment:
`atom.config.set('linter-phpmd.rulesOrConfigFile', atom.config.get('linter-phpmd.rulesets'));`

We want to save their old setting, not grab it this one time and then lose it forever 😉.

Reply:
Doh, you're right 🙄  where's my head at? 😜